src/js/todolist.js

```diff
@@ -0,0 +1,150 @@
+import Task from './task.js';
+import threeDots from '../images/three-dots.png';
+import domElements from './dom_elements.js';
+
+class TodoList {
+  constructor(listElem) {
+    this.list = [];
+    this.listElem = listElem;
+  }
+
+  init() {
+    if (JSON.parse(localStorage.getItem('list')) !== null) {
+      this.list = JSON.parse(localStorage.getItem('list'));
+      this.list.forEach((element) => {
+        this.insertToDocument(new Task(element.index, element.completed, element.description));
+      });
+    }
+  }
+
+  addTask(completed, description) {
+    const { length } = this.list;
+    const task = new Task(length, completed, description);
+    this.list.push(task);
+    this.insertToDocument(task);
+    this.saveToLocalStorage();
+  }
+
+  insertToDocument(task) {
+    const elem = document.createElement('li');
+    elem.id = `item${task.index}`;
+    elem.classList.add('task');
+    elem.innerHTML += `<input type="checkbox" name='completed' value='${task.completed ? 'completed' : 'not-completed'}' ${task.completed ? 'checked' : ''}>
+    <span>${task.description}</span>
+    <img src=${threeDots} alt="Three Dots Button">`;
+    this.listElem.append(elem);
+    this.addEventListenerToListItem(elem, task.index);
+  }
+
+  deleteTask(index) {
+    this.sortTasks();
+    let newIndex = 0;
+    const newList = [];
+    this.list.forEach((task) => {
+      if (index !== task.index) {
+        newList.push(new Task(newIndex, task.completed, task.description));
+        newIndex += 1;
+      }
+    });
+    this.list = newList;
+    this.saveToLocalStorage();
+    TodoList.updateDOM(index);
+  }
+
+  static updateDOM(index) {
+    const allTaskItems = document.querySelectorAll('.task');
+    let newIndex = 0;
+    allTaskItems.forEach((listItem) => {
+      if (listItem.getAttribute('id') === `item${index}`) {
+        listItem.remove();
+      } else {
+        listItem.setAttribute('id', `item${newIndex}`);
+        newIndex += 1;
+      }
+    });
+  }
+
+  editTaskDescription(index, description) {
+    this.list[index].description = description;
+    this.saveToLocalStorage();
+  }
```

Comment:
- [ ] A task should be editable. Kindly ensure that this functionality is implemented as I was not able to edit a task